src/collapse/vsprof.rs

```diff
@@ -0,0 +1,222 @@
+use std::{
+    cmp::Ordering,
+    io::{self, BufRead},
+};
+
+use log::warn;
+
+use crate::collapse::{common::Occurrences, Collapse};
+
+static START_LINE: &str = "Level,Function Name,Number of Calls,Elapsed Inclusive Time %,Elapsed Exclusive Time %,Avg Elapsed Inclusive Time,Avg Elapsed Exclusive Time,Module Name,";
+
+/// A stack collapser for the output of the Visual Studio built in profiler.
+#[derive(Default)]
+pub struct Folder {
+    /// Function entries on the stack in this entry thus far.
+    stack: Vec<(String, usize)>,
+}
+
+impl Collapse for Folder {
+    fn collapse<R, W>(&mut self, mut reader: R, writer: W) -> io::Result<()>
+    where
+        R: std::io::BufRead,
+        W: std::io::Write,
+    {
+        // Skip the header
+        let mut line = Vec::new();
+        if reader.read_until(b'\n', &mut line)? == 0 {
+            warn!("File ended before start of call graph");
+            return Ok(());
+        };
+
+        let header = String::from_utf8_lossy(&line).to_string();
+        if !line_matches_start_line(header.clone()) {
+            return invalid_data_error!("Incorrect header:\n{}", header);
```

Comment:
Let's be a little more helpful here. How about
```suggestion
            return invalid_data_error!("Expected first line to be header line\n    {}\nbut instead got\n    {}", START_LINE, header);
```